src/mlpack/core/data/load_csv.hpp

```diff
@@ -210,32 +315,62 @@ class LoadCSV
       // If we need to do a first pass for the DatasetMapper, do it.
       if (MapPolicy::NeedsFirstPass)
       {
+        // In this case we must pass everything we parse to the MapPolicy.
         size_t dim = 0;
 
-        // In this case we must pass everything we parse to the MapPolicy.
-        auto firstPassMap = [&](const iter_type& iter)
-        {
-          std::string str(iter.begin(), iter.end());
-          boost::trim(str);
+	std::stringstream line_stream;
+	std::string token;
 
-          info.template MapFirstPass<T>(std::move(str), dim++);
-        };
+	if(line.size() == 0)
+	{
+	  break;
+	}
 
-        // Now parse the line.
-        qi::parse(line.begin(), line.end(),
-            stringRule[firstPassMap] % delimiterRule);
+	line_stream.clear();
+	line_stream.str(line);
+
+	while(line_stream.good())
+        {
+          std::getline(line_stream, token, delim);
+	  // Remove whitespace from either side
+          trim(token);
+  
+          if(token[0] == '"' && token[token.size() - 1] != '"')
+          {
+	    std::string tok = token;
+
+	    while(token[token.size() - 1] != '"')
+	    {
+	      tok += delim;
+	      std::getline(line_stream, token, delim);
+	      tok += token;
+            }
+
+	    token = tok;
+          }
```

Comment:
Since this is used twice, it might be useful to refactor it out into a separate utility function. :+1:

Reply:
Done!!